manifests/fpm.pp

```diff
@@ -45,21 +45,21 @@
 #
 class php::fpm (
   $ensure               = $::php::ensure,
-  $service_ensure       = $::php::params::fpm_service_ensure,
-  $service_enable       = $::php::params::fpm_service_enable,
-  $service_name         = $::php::params::fpm_service_name,
-  $service_provider     = undef,
+  $service_ensure       = $::php::fpm_service_ensure,
+  $service_enable       = $::php::fpm_service_enable,
+  $service_name         = $::php::fpm_service_name,
+  $service_provider     = $::php::fpm_service_provider,
   $package              = "${::php::package_prefix}${::php::params::fpm_package_suffix}",
   $inifile              = $::php::params::fpm_inifile,
-  $settings             = {},
+  $settings             = $::php::real_settings,
   $global_pool_settings = {},
   $pools                = { 'www' => {} },
-  $log_owner            = $::php::params::fpm_user,
-  $log_group            = $::php::params::fpm_group,
-) inherits ::php::params {
+  $log_owner            = $::php::log_owner,
+  $log_group            = $::php::log_group,
+) {
 
   if $caller_module_name != $module_name {
-    warning('php::fpm is private')
+    fail('php::fpm is private')
```

Comment:
I guess all of the parameters should be exposed in the base class, otherwise you *have* to use hiera to set them.  What about switching to an `if ! defined(Class['php'])` here too?  If you really wanted to declare `php::fpm` yourself, you still could as long as you declared `php` first with `fpm => false`.

It's going to be a new major release, and the class was marked as private, so people can't complain too much if we make a backwards incompatible change?  After all, the class was actively warning them not to declare it themselves!